salt/states/heat.py

```diff
@@ -0,0 +1,342 @@
+    return tpl
+
+
+def deployed(name, template=None, enviroment=None, params=None, poll=5,
+             rollback=False, timeout=3600, update=False, profile=None,
+             **connection_args):
+    '''
+    Deploy stack with the specified properties
+
+    name
+        The name of the stack
+
+    template
+        File of template
+
+    enviroment
+        File of enviroment
+
+    params
+        Parameter dict used to create the stack
+
+    poll
+        Poll(in sec.) and report events until stack complete
+
+    rollback
+        Enable rollback on create failure
+
+    timeout
+        Stack creation timeout in minutes
+
+    profile
+        Profile to use
+
+    '''
+    log.debug('Deployed with(' +
+              '{0}, {1}, {2}, {3}, {4}, {5}, {6}, {7}, {8}, {9})'
+              .format(name, template, enviroment, params, poll, rollback,
+                      timeout, update, profile, connection_args))
+    ret = {'name': None,
+           'comment': '',
+           'changes': {},
+           'result': True}
+
+    if not name:
+        ret['result'] = False
+        ret['comment'] = 'Name ist not valid'
+        return ret
+
+    ret['name'] = name,
+
+    existing_stack = __salt__['heat.show_stack'](name, profile=profile)
+
+    if existing_stack['result'] and not update:
+        ret['comment'] = 'Stack {0} is deployed'.format(name)
+        return ret
+    if existing_stack['result'] and update:
+        if template:
+            template_tmp_file = salt.utils.mkstemp()
+            tsfn, source_sum, comment_ = __salt__['file.get_managed'](
+                name=template_tmp_file,
+                template=None,
+                source=template,
+                source_hash=None,
+                user=None,
+                group=None,
+                mode=None,
+                saltenv='base',
+                context=None,
+                defaults=None,
+                skip_verify=False,
+                kwargs=None)
+
+            template_manage_result = __salt__['file.manage_file'](
+                name=template_tmp_file,
+                sfn=tsfn,
+                ret=None,
+                source=template,
+                source_sum=source_sum,
+                user=None,
+                group=None,
+                mode=None,
+                saltenv='base',
+                backup=None,
+                makedirs=True,
+                template=None,
+                show_changes=False,
+                contents=None,
+                dir_mode=None)
+
+            if (template_manage_result['result']) or \
+                    ((__opts__['test']) and (template_manage_result['result'] is not False)):
+                with salt.utils.fopen(template_tmp_file, 'r') as tfp_:
+                    tpl = tfp_.read()
+                    salt.utils.safe_rm(template_tmp_file)
+                    try:
+                        if isinstance(tpl, six.binary_type):
+                            tpl = tpl.decode('utf-8')
```

Comment:
In heatclient is checking template on utf-8, think that responses back aren't in non-utf-8.